hathor/p2p/sync_v2/manager.py

```diff
@@ -0,0 +1,1207 @@
+        # transactions in the case when I am downloading a side chain. Starts at the genesis, which is common to all
+        # peers on the network
+        self._last_received_block: Optional[Block] = None
+
+        # Saves if I am in the middle of a mempool sync
+        # we don't execute any sync while in the middle of it
+        self.mempool_manager = SyncMempoolManager(self)
+        self._receiving_tips: Optional[list[bytes]] = None
+
+        # Cache for get_tx calls
+        self._get_tx_cache: OrderedDict[bytes, BaseTransaction] = OrderedDict()
+        self._get_tx_cache_maxsize = 1000
+
+        # Looping call of the main method
+        self._lc_run = LoopingCall(self.run_sync)
+        self._lc_run.clock = self.reactor
+        self._is_running = False
+
+        # Whether we propagate transactions or not
+        self._is_relaying = False
+
+        # This stores the final height that we expect the last "get blocks" stream to end on
+        self._blk_end_height: Optional[int] = None
+
+        # Whether to sync with this peer
+        self._is_enabled: bool = False
+
+    def get_status(self) -> dict[str, Any]:
+        """ Return the status of the sync.
+        """
+        res = {
+            'is_enabled': self.is_sync_enabled(),
+            'peer_height': self.peer_height,
+            'synced_height': self.synced_height,
+            'synced': self._synced,
+            'state': self.state.value,
+        }
+        return res
+
+    def is_synced(self) -> bool:
+        return self._synced
+
+    def is_errored(self) -> bool:
+        return self.state is PeerState.ERROR
+
+    def is_sync_enabled(self) -> bool:
+        return self._is_enabled
+
+    def enable_sync(self) -> None:
+        self._is_enabled = True
+
+    def disable_sync(self) -> None:
+        self._is_enabled = False
+
+    def send_tx_to_peer_if_possible(self, tx: BaseTransaction) -> None:
+        if not self._is_enabled:
+            self.log.debug('sync is disabled')
+            return
+        if not self.is_synced():
+            # XXX Should we accept any tx while I am not synced?
+            return
+
+        # XXX When we start having many txs/s this become a performance issue
+        # Then we could change this to be a streaming of real time data with
+        # blocks as priorities to help miners get the blocks as fast as we can
+        # We decided not to implement this right now because we already have some producers
+        # being used in the sync algorithm and the code was becoming a bit too complex
+        if self._is_relaying:
+            self.send_data(tx)
+
+    def is_started(self) -> bool:
+        return self._started
+
+    def start(self) -> None:
+        """ Start sync.
+        """
+        if self._started:
+            raise Exception('NodeSyncBlock is already running')
+        self._started = True
+        self._lc_run.start(5)
```

Comment:
Would there be any penalty for us if we decrease this time from 5s to maybe 1s or 2s?

I have the impression we could be losing some seconds between the end of the current streaming and starting the next one.

Reply:
We might just move it to a variable and let us change it through syscall.